wasm/src/malloc.c

```diff
@@ -98,16 +134,95 @@ void opa_heap_top_set(unsigned int top)
     init_free();
 }
 
+// returns the free list applicable for the requested size.
+static struct heap_blocks * __opa_blocks(size_t size) {
+    for (int i = 0; i < sizeof(heap_free) / sizeof(struct heap_blocks); i++) {
```

Comment:
Not sure if it would make sense for loop alone, but let me think I can somehow enforce more compile time checks for the initialization of the lists as well (to guarantee they are all initialized). That might require introducing some macros.